src/lib/Config.ts

```diff
@@ -0,0 +1,116 @@
+type SpecValue =
```

Comment:
Personally, I find this file rather confusing. It seems like it is parsing, validating, and providing overrides and default values all at once. Am I understanding it correctly? If so, what would you say to splitting those things up into separate, single-purpose functions which can be chained?

Reply:
I rewrote this using [ajv](https://ajv.js.org)